### PR TITLE
Add serialization trait bounds to `Polynomial`

### PR DESCRIPTION
**Pull request by ryanleh**

### CODE REVIEW

**Comment by Pratyush** [comment]
Hmm for the time being I would implement it just for `UnivariatePolynomial`; I don't think it makes sense to implement for multivariate polynomials just yet. I would also not include it as trait bound here; instead let's just add the bounds at the use sites in the delegation project.

**Comment by ryanleh** [comment]
Ok, I just derived it for `DensePolynomial`

**Comment by ryanleh** [comment]
Or wait, I think it will need to be a bound on `UVPolynomial` for it to work with the structs in `poly-commit`

File: poly/src/polynomial/univariate/sparse.rs [hunk 1]
```diff
@@ -2,10 +2,11 @@
 use crate::univariate::{DenseOrSparsePolynomial, DensePolynomial};
 use crate::{EvaluationDomain, Evaluations, UVPolynomial};
 use ark_ff::{FftField, Field};
+use ark_serialize::*;
 use ark_std::{collections::BTreeMap, fmt, vec::Vec};
 
 /// Stores a sparse polynomial in coefficient form.
-#[derive(Clone, PartialEq, Eq, Hash, Default)]
+#[derive(Clone, PartialEq, Eq, Hash, Default, CanonicalSerialize, CanonicalDeserialize)]
```
Comment (Pratyush) [inline poly/src/polynomial/univariate/sparse.rs:9]: Does this need to be serializable?

Comment (ryanleh) [reply poly/src/polynomial/univariate/sparse.rs:9]: In order for the `marlin_pc::Randomness` struct to be serializable, you need `UVPolynomial: CanonicalSerialize + CanonicalDeserialize` so ya.

Comment (Pratyush) [reply poly/src/polynomial/univariate/sparse.rs:9]: but `SparsePolynomial` doesn't implement `UVPolynomial`, no? Or am I not finding the right impl?

(I don't see `DenseOrSparsePolynomial` in the diff, so I assume you're not deriving it for that)

Comment (ryanleh) [reply poly/src/polynomial/univariate/sparse.rs:9]: Oh good point haha I forgot about that - will adjust.

**Pull request merged**